datafusion/core/src/datasource/listing/helpers.rs

```diff
@@ -99,6 +99,7 @@ impl ExpressionVisitor for ApplicabilityVisitor<'_> {
             Expr::ScalarUDF { fun, .. } => {
                 self.visit_volatility(fun.signature.volatility)
             }
+            Expr::TableUDF { fun, .. } => self.visit_volatility(fun.signature.volatility),
```

Comment:
I recommend writing it like this:
```rust
            Expr::ScalarUDF { fun, .. } | Expr::TableUDF { fun, .. } => {
                self.visit_volatility(fun.signature.volatility)
            }
```

Reply:
good point, however, it doesn't work in this case (because argument fun has different types for TableUDF and ScalarUDF)